File: JavaScript/Edge of the Ocean/adjacentElementsProduct.js
const adjacentElementsProduct = inputArray => {
    let greatest = inputArray[0] * inputArray[1];
    for (let i = 1; i < inputArray.length - 1; i++) {
        if (inputArray[i] * inputArray[i + 1] > greatest) greatest = inputArray[i] * inputArray[i + 1];
    }
    return greatest;
};

File: CodeSignal/Edge of the Ocean/go - 5:5/adjacentElementsProduct.go
func adjacentElementsProduct(inputArray []int) int {
    greatest := inputArray[0] * inputArray[1];
    for i := 1; i < len(inputArray) - 1; i++ {
        nextProd := inputArray[i] * inputArray[i + 1]
        if nextProd > greatest {
            greatest = nextProd
        }
    }
    return greatest
}
